vm.go

```diff
@@ -584,11 +582,6 @@ func (vm *vm) run() {
 		} else {
 			vm.prg.code[vm.pc].exec(vm)
 		}
-		ticks++
-		if ticks > 10000 {
-			runtime.Gosched()
```

Comment:
are there any concerns with removing this?

Reply:
Hmm wouldn't we want to still have this around? 

I thought that this is our attempt at guaranteeing fairness across all VMs running on the pod.

maybe it does introduce extra latency because of context switching but I feel like this might still be worth keeping around?

Reply:
I was looking at https://github.com/dop251/goja/issues/97 which is why this change was in. I haven't taken a deeper look but as Calvin mentioned the test for this was commented out and it's currently failing so we might be OK removing this. That said it seems like the improvement is minimal compared to other changes so I'm not opposed to keeping this around.

Reply:
From the 10m pprof dump Andy [linked](https://mongodb.slack.com/archives/C0588TTRPBJ/p1691174748040029) in slack it looks like we would see a 39% improvement to the flat .run time.

```
    153.43s   1769.57s (flat, cum) 54.00% of Total
    ...
    57.58s     57.58s    589:		ticks++
     2.80s      2.80s    590:		if ticks > 10000 {
         .          .    591:			runtime.Gosched()
         .          .    592:			ticks = 0
         .          .    593:		}
```

Reply:
I'm going to move that preempt test into baas to see if we get this for free from our added interrupt logic (memoryTracking).

Worst case scenario, I'd say we could call runtime.Gosched() on halt of the function or even let it correspond with our limiterTicksLeft check and call runtime.Gosched() near the `self.fillBucket()` call.

Reply:
> see if we get this for free from our added interrupt logic (memoryTracking) or in our waitOneTick method

To provide more details on this, calling time.Sleep() will also yield to the CPU, there are instances in golang source code where time.sleep(Nanosecond) is used instead of calling runtime.gosched(). So our [waitN](https://github.com/Calvinnix/goja/blob/bb8a7c8db228f8aeb17d400a7d8eb90f90325da3/limiter.go#L39) might be a spot where this is happening already.

Reply:
> I'm going to move that preempt test into baas to see if we get this for free from our added interrupt logic (memoryTracking) or in our waitOneTick method.

Ok it ended up being more direct than relying on some obscure go functionality to yield to the CPU. In the loop we are [breaking](https://github.com/Calvinnix/goja/blob/f5cc0e5207eef6e780d47a7cc42a7bb936244045/vm.go#L555) if we get an interrupt that isn't of type function and [later we panic if it is an error](https://github.com/Calvinnix/goja/blob/f5cc0e5207eef6e780d47a7cc42a7bb936244045/vm.go#L606) so this is why we had that preempt test commented out. 

Not that it matters too much because we've diverged a bit from current goja, but this tick counter does not exist in the current version of goja, it was [removed here](https://github.com/Calvinnix/goja/commit/0cb3e9b4a7a835579e34ccefb7a45dd893f81e83#diff-687ac236cd82f74130ba2a4fcba7a9e157497ffdecad9e5643a1ac1d37d9b08fL495-L500).

Reply:
Ok I went ahead and confirmed this with trace.

I have 2 goroutines running in parallel and we can see in the screenshot below that they both share the thread (limited to one thread via `runtime.GOMAXPROCS(1)`)
![image](https://github.com/mongodb-forks/goja/assets/6693710/2ef6375b-2eae-42f3-b5fa-de016dc91283)

As a nice aside, removing the runtime.gosched call is also nice since it looks like we were swapping context too often. (Notice the time labels at the top) 
![image](https://github.com/mongodb-forks/goja/assets/6693710/8df112d6-052c-4295-8e16-119f53070693)